File: backend/app/models/event.py
from __future__ import annotations

from datetime import UTC, datetime

from sqlalchemy import JSON, DateTime, Index, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models import Base


class Event(Base):
    """关键服务事件 —— 承载启动 / 部署 / 启动失败等业务事件。

    区别于 ``Log`` 表：log 是机器噪音/WARNING+ 持久化；event 是给人看的业务事件。
    """

    __tablename__ = "event"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    timestamp: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(UTC),
        index=True,
    )
    type: Mapped[str] = mapped_column(String(50), index=True)
    source: Mapped[str] = mapped_column(String(100))
    title: Mapped[str] = mapped_column(String(255))
    message: Mapped[str] = mapped_column(Text)
    extra: Mapped[dict] = mapped_column(JSON, default=dict)

    __table_args__ = (Index("ix_event_type_timestamp", "type", "timestamp"),)

File: go-backend/internal/model/event.go
package model

import (
	"time"

	"gorm.io/datatypes"
)

// Event 承载关键服务事件（启动 / 部署 / 启动失败等业务事件）。
//
// 区别于 Log 表：log 是机器噪音/WARNING+ 持久化；event 是给人看的业务事件。
// 字段与索引对齐 Python backend/app/models/event.py 的 SQLAlchemy 定义，
// 确保 GORM AutoMigrate 生成的约束/索引名与 Alembic 已有的命名一致。
type Event struct {
	ID        uint           `gorm:"primaryKey;index"`
	Timestamp time.Time      `gorm:"index:ix_event_type_timestamp,priority:2;default:current_timestamp"`
	Type      string         `gorm:"size:50;index:ix_event_type_timestamp,priority:1"`
	Source    string         `gorm:"size:100"`
	Title     string         `gorm:"size:255"`
	Message   string         `gorm:"type:text"`
	Extra     datatypes.JSON `gorm:"type:jsonb"`
}

// TableName 对齐 Python SQLAlchemy 的 __tablename__ = "event"。
func (Event) TableName() string {
	return "event"
}
